packages/core/src/utils/transformers/transformImageAsset.ts

```diff
@@ -0,0 +1,67 @@
+import { SupportedImageFormats } from '@/constants';
+import { OptimizedImageAsset } from '@/types';
+import { AssetFile } from 'contentful';
+
+interface AssetFileWithRequiredImage extends AssetFile {
+  details: Required<AssetFile['details']>;
+}
+type ValidFormats = (typeof SupportedImageFormats)[number];
+
+const MAX_WIDTH_ALLOWED = 4000;
+
+export const transformImageAsset = (
+  file: AssetFile,
+  sizes?: string,
+  quality: number = 100,
```

Comment:
Should this be an `options` object, where `quality` is just one possible factor?